thoth/storages/graph/postgres.py

```diff
@@ -908,24 +1039,111 @@ def get_unsolved_python_packages_all_versions(
             python_version=python_version
         )
 
-        query = query.offset(start_offset).limit(count)
+        query = query.offset(start_offset).limit(count/2)
 
         if distinct:
             query = query.distinct()
 
-        query = query.all()
+        result = query.all()
 
         query_result = {}
 
-        for item in query:
-            if item[0] not in query_result.keys():
+        for item in result:
+            if item[0] not in query_result:
                 query_result[item[0]] = []
+            query_result[item[0]].append((item[1], item[2]))
+
+        unsolved = self._get_unsolved_python_package_versions_count_edge_cases()
+
+        if unsolved and count > 0:
```

Comment:
This is kind-of weird behavior

Reply:
I adjusted it, WDYT?

Reply:
In this way, it is more clear, we are splitting in two and we have +- 1 in count.